### DISCUSSION TITLE
Sending ADC value to another cc2650 launchpad using TSCH

### DISCUSSION THREAD

**Discussion by carolinapinho** [contiki-ng/contiki-ng#1820] [Q&A]
Hello! I am doing a project where I want to use IEEE 802.15.4 with TSCH and I have already been able to execute the examples given in the repository, the hello-world, and the platform-specific (cc2650) examples with TSCH. With this, I certified that I can make my boards associate and communicate (ping), making one the coordinator and the other the associated node using RPL. My next step is to make one node send some information of my choice via TSCH|RPL (for example the value of an ADC or a simple on/off signal) to the other node, however, I can not understand where do I need to start. I read somewhere that I need to write in some specific registers but i cant find more information about this. 
Can someone please point me in the right direction?  
Thank you for your attention.

**Comment by atiselsts** [comment]
I assume that you already have the value you want to send, and you're not looking for guidance on reading the ADC. 

Normally one does not send data using RPL or TSCH directly. You would use either a transport layer protocol like UDP or application layer protocol like CoAP, MQTT or HTTP. For all of these we have examples in Contiki-NG, in:

- `examples/rpl-udp/`
- `examples/coap`
- `examples/mqtt-client/`
- `examples/platform-specific/cc26x0-cc13x0/cc26x0-web-demo/`

See this tutorial on how to switch an existing example to use TSCH: https://github.com/contiki-ng/contiki-ng/wiki/Tutorial%3A-Switching-to-TSCH.

**Reply by carolinapinho** [reply]
Hi! @atiselsts  Thank you for your answer!
My objective is not to use neither of those upper layers because I want to send short messages and protocols like UDP would introduce overhead that I don't want. I was looking for exchange messages at the 6lowpan level, is that possible with Contiki's existing APIs?

**Reply by atiselsts** [reply]
UDP header is just 8 bytes, and the 6LoWPAN header compression mechanism can even compress the UDP header further. You should now expect large benefits from using 6LoWPAN without a transport protocol.

That said, and I don't think there any ready made examples, you're of course able to create your own IP packets without the UDP header, and then calling `tcpip_ipv6_output()`.

**Reply by Mbaron12** [reply]
> I assume that you already have the value you want to send, and you're not looking for guidance on reading the ADC.
> 
> Normally one does not send data using RPL or TSCH directly. You would use either a transport layer protocol like UDP or application layer protocol like CoAP, MQTT or HTTP. For all of these we have examples in Contiki-NG, in:
> 
> * `examples/rpl-udp/`
> * `examples/coap`
> * `examples/mqtt-client/`
> * `examples/platform-specific/cc26x0-cc13x0/cc26x0-web-demo/`
> 
> See this tutorial on how to switch an existing example to use TSCH: https://github.com/contiki-ng/contiki-ng/wiki/Tutorial%3A-Switching-to-TSCH.

Hi sir ! I want to ask you some questions about tsch .
IF the rpl-border-router mote use tsch ,the rpl-border-router mote can work right ? My rpl-border-route mote which use tsch can not work right.
How can I solve this problem? Looking forward to your reply!

contiki-ng-develop/examples/rpl-border-router-1$ sudo make TARGET=cc2538dk connect-router
sudo ../../tools/serial-io/tunslip6 fd00::1/64
********SLIP started on /dev/ttyUSB0'' opened tun device /dev/tun0''
ifconfig tun0 inet hostname mtu 1500 up
ifconfig tun0 add fd00::1/64
ifconfig tun0 add fe80::0:0:0:1/64
ifconfig tun0

tun0: flags=4305<UP,POINTOPOINT,RUNNING,NOARP,MULTICAST> mtu 1500
inet 127.0.1.1 netmask 255.255.255.255 destination 127.0.1.1
inet6 fe80::b4b9:8473:81c5:2802 prefixlen 64 scopeid 0x20
inet6 fd00::1 prefixlen 64 scopeid 0x0
inet6 fe80::1 prefixlen 64 scopeid 0x20
unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00 txqueuelen 500 (UNSPEC)
RX packets 0 bytes 0 (0.0 B)
RX errors 0 dropped 0 overruns 0 frame 0
TX packets 0 bytes 0 (0.0 B)
TX errors 0 dropped 0 overruns 0 carrier 0 collisions 0

**Comment by Mbaron12** [comment]
Hi sir ! I want to ask you some questions about  tsch .
IF the rpl-border-router mote use tsch ,the  rpl-border-router mote can work right ?  My rpl-border-route  mote which use tsch can not  work right.
How can I solve this problem? Looking forward to your reply!
### contiki-ng-develop/examples/rpl-border-router-1$ sudo make TARGET=cc2538dk connect-router
sudo ../../tools/serial-io/tunslip6 fd00::1/64
********SLIP started on ``/dev/ttyUSB0''
opened tun device ``/dev/tun0''
ifconfig tun0 inet `hostname` mtu 1500 up
ifconfig tun0 add fd00::1/64
ifconfig tun0 add fe80::0:0:0:1/64
ifconfig tun0

tun0: flags=4305<UP,POINTOPOINT,RUNNING,NOARP,MULTICAST>  mtu 1500
        inet 127.0.1.1  netmask 255.255.255.255  destination 127.0.1.1
        inet6 fe80::b4b9:8473:81c5:2802  prefixlen 64  scopeid 0x20<link>
        inet6 fd00::1  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::1  prefixlen 64  scopeid 0x20<link>
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 500  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
